src/get-started/java-binding.md

```diff
@@ -401,39 +443,51 @@ An exception of any other type will be recorded with _no_ error code
 as an "unexpected" execution error.
 
 If transaction execution fails, the changes made by the transaction are
-rolled back, while the error data is stored in the database for further user
-reference. Light clients also provide access to information on the
-transaction execution result (which may be either success or failure)
+rolled back, while the error data is [stored in the database][call-errors-registry]
+for further user reference. Light clients also provide access to information
+on the transaction execution result (which may be either success or failure)
 to their users.
 
-#### After Transactions Handler
+[execution-exception]: https://exonum.com/doc/api/java-binding/0.10.0/com/exonum/binding/core/transaction/ExecutionException.html
+[call-errors-registry]: https://exonum.com/doc/api/java-binding/0.10.0/com/exonum/binding/core/blockchain/Blockchain.html#getCallErrors(long)
 
-Services can receive a notification after all transactions in a block has been
-processed but before the block has been committed, allowing inspecting
-the changes made by the transactions and modifying any service state.
+#### Before and After Transactions Handlers
 
-Exonum delivers this notification to implementations
-of [`Service#beforeCommit`][service-before-commit] method. The method will see
-the state of the database after all transactions have been applied.
-It may make any changes to the service state, which will be included
+Services can receive notifications:
+
+- before any transactions in a block has been processed
+- after all transactions in a block has been processed, but before the block 
+has been committed.
```

Comment:
Nit: Incorrect Markdown formatting; this line should be indented with 2 spaces to clearly make it a part of a list item.